src/sensors/include/iDynTree/Sensors/PredictSensorsMeasurements.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2015 Fondazione Istituto Italiano di Tecnologia
+ * Author: Naveen Kuppuswamy
+ * email:  naveen.kuppuswamy@iit.it
+ * Permission is granted to copy, distribute, and/or modify this program
+ * under the terms of the GNU General Public License, version 2 or any
+ * later version published by the Free Software Foundation.
+ *
+ * A copy of the license can be found at
+ * http://www.robotcub.org/icub/license/gpl.txt
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details
+ */
+
+#ifndef PREDICTSENSORSMEASUREMENTS_HPP
+#define PREDICTSENSORSMEASUREMENTS_HPP
+
+
+namespace iDynTree{
+    
+    class SensorsList;
+    class Traversal;
+    class Model;
+    class VectorDynSize;
+
+    class PredictSensorsMeasurements {
+    
+        
+    public : 
+        PredictSensorsMeasurements();
+        bool setSensorList(const SensorsList &sensorsList);
+        bool setModel(const iDynTree::Model &model);
+        bool setTraversal(const iDynTree::Traversal &traversal);
+        bool makePrediction(iDynTree::VectorDynSize &predictedMeasurement);
+        ~PredictSensorsMeasurements();
+        
+    private :
+        
+        struct PredictSensorsMeasurementsPrivateAttributes;
+        PredictSensorsMeasurementsPrivateAttributes * pimpl;
```

Comment:
I tried to keep it that way because its cleaner and its unnecessary for the
user to understand that it encodes a Model, Traversal and a SensorList. On
the other hand, that is made obvious by the forward declarations..Anyways,
will code in the mapping function to the ForwardKinematics and organise my
work in a few pushes.


On Sun, Nov 29, 2015 at 6:10 PM, Silvio Traversaro <notifications@github.com
> wrote:

> In src/sensors/include/iDynTree/Sensors/PredictSensorsMeasurements.hpp
> <https://github.com/robotology/idyntree/pull/104#discussion_r46094980>:
>
> > +
> > +    class PredictSensorsMeasurements {
> > +
> > +
> > +    public :
> > +        PredictSensorsMeasurements();
> > +        bool setSensorList(const SensorsList &sensorsList);
> > +        bool setModel(const iDynTree::Model &model);
> > +        bool setTraversal(const iDynTree::Traversal &traversal);
> > +        bool makePrediction(iDynTree::VectorDynSize &predictedMeasurement);
> > +        ~PredictSensorsMeasurements();
> > +
> > +    private :
> > +
> > +        struct PredictSensorsMeasurementsPrivateAttributes;
> > +        PredictSensorsMeasurementsPrivateAttributes * pimpl;
>
> I introduced the PIMPL in some classes to avoid having to include legacy
> KDL class in the headers , but in this case I guess you can also avoid it,
> as you prefer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robotology/idyntree/pull/104/files#r46094980>.
>